components/resource_metering/src/reporter/mod.rs

```diff
@@ -89,27 +90,25 @@ impl Reporter {
 
             data_sinks: HashMap::default(),
             records: Records::default(),
+            region_records: RegionRecords::default(),
         }
     }
 
     fn handle_records(&mut self, records: Arc<RawRecords>) {
         let ts = records.begin_unix_time_secs;
+        let n = self.config.max_resource_groups;
         let agg_map = records.aggregate_by_extra_tag();
```

Comment:
Maybe we can try to aggregate together to reduce one loop if there are too many data.